ex2_system/source/beacon/beacon_task.c

```diff
@@ -32,13 +32,33 @@ SAT_returnState start_beacon_daemon(void);
  * @param pvParameters
  *    task parameters (not used)
  */
-static void * beacon_daemon(void *pvParameters) {
-    TickType_t delay = pdMS_TO_TICKS(1000);
-    for ( ;; ) {
-        // TODO construct and send the system beacon
+static void *beacon_daemon(void *pvParameters) {
+  TickType_t delay = pdMS_TO_TICKS(1000);  // make this config.able
+  for (;;) {
+    int8_t uhf_status = -1;
+    /* Constructing the system beacon content */
+    // Refer to table 3 of MOP
+    UHF_configStruct beacon_msg;
+    // TODO: call the appropriate HAL functions to get the most updated or
+    // cached information of the components + state machine, RTC, etc.
 
-        vTaskDelay(delay);
+    uhf_status = HAL_UHF_setBeaconMsg(beacon_msg);
+
+    /* Sending the beacon */
+    // The beacon transmission period is configurable through comms service
+    // by the operator or here through HAL_UHF_getBeaconT().
+    uint8_t scw[SCW_LEN];
+    uhf_status = HAL_UHF_getSCW(scw);
+    if (uhf_status == U_GOOD_CONFIG) {  // replace 0 with no error status
+      scw[5] = 1;
+      uhf_status = HAL_UHF_setSCW(scw);
+    }
+    if (uhf_status != 0) {  // If get or set scw fail
+                            // Change delay? reset TRX?

```

Comment:
Do we really have any plan for when a component is turned off when it's not supposed to?